packages/payment-mop-up-job/src/processors/processor.js

```diff
@@ -18,6 +18,34 @@ const limiter = new Bottleneck({
   maxConcurrent: process.env.CONCURRENCY || CONCURRENCY_DEFAULT
 })
 
+const getPaymentJournalStatusCodeFromGovPayErrorStatusCode = govPayErrorStatusCode => {
+  switch (govPayErrorStatusCode) {
+    case GOVUK_PAY_ERROR_STATUS_CODES.EXPIRED:
+    case GOVUK_PAY_ERROR_STATUS_CODES.NOT_FOUND:
+      return PAYMENT_JOURNAL_STATUS_CODES.Expired
+    case GOVUK_PAY_ERROR_STATUS_CODES.USER_CANCELLED:
+      return PAYMENT_JOURNAL_STATUS_CODES.Cancelled
+    case GOVUK_PAY_ERROR_STATUS_CODES.REJECTED:
+      return PAYMENT_JOURNAL_STATUS_CODES.Failed
+  }
+}
+
+const shouldUpdatePaymentJournal = transaction => {
+  const { code, state } = transaction.paymentStatus
+  const { paymentTimestamp } = transaction
+
+  const isExpiredCancelledOrRejected = [
+    GOVUK_PAY_ERROR_STATUS_CODES.EXPIRED,
+    GOVUK_PAY_ERROR_STATUS_CODES.USER_CANCELLED,
+    GOVUK_PAY_ERROR_STATUS_CODES.REJECTED
+  ].includes(state?.code)
+
+  const isExpiredNotFound =
```

Comment:
Can we keep the old `// The payment's not found and three hours have elapsed` comment here? Wasn't entirely sure what circumstances this was for at first